src/main/java/firebase/sdk/dashboard/dao/SDKDaoDatastore.java

```diff
@@ -0,0 +1,343 @@
+package firebase.sdk.dashboard.dao;
+
+import firebase.sdk.dashboard.data.SDK;
+import firebase.sdk.dashboard.data.SDKReleaseMetadata;
+import firebase.sdk.dashboard.data.VersionMetadata;
+import firebase.sdk.dashboard.data.Release;
+import firebase.sdk.dashboard.data.Platform;
+import java.util.List;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+import java.time.Instant;
+import java.io.IOException;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.KeyFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.EmbeddedEntity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.FilterPredicate;
+import com.google.appengine.api.datastore.Query.FilterOperator;
+import com.google.appengine.api.datastore.Query.CompositeFilter;
+import com.google.appengine.api.datastore.Query.CompositeFilterOperator;
+import com.google.appengine.api.datastore.FetchOptions;
+import com.google.appengine.api.datastore.QueryResultIterable;
+
+/**
+ * An interface for retrieving and manipulating data about SDKs.
+ */
+public class SDKDaoDatastore implements SDKDao {
+
+  private static final DatastoreService DATASTORE = DatastoreServiceFactory.getDatastoreService();
+
+  public SDK getSDK(Platform givenPlatform, String givenLibraryName) {
```

Comment:
givenPlatform -> platform, since its obvious that it is given :)
Also, givenLibraryName -> libraryName, or rather sdkName, coz we use SDK as the main term.

Reply:
I added given to the front of the parameters because later in the function i create variables that use those names. I can change givenLibraryName to sdkName though

Reply:
Actually I decided to have the parameter names as the one you recommended and changed the other variable names that were within the function